projects/pswg/src/main/java/com/parzivail/pswg/entity/ThermalDetonatorEntity.java

```diff
@@ -77,30 +61,25 @@ public void tick()
 	}
 
 	@Override
-	protected void createParticles(ServerWorld world, double x, double y, double z)
+	protected void createParticles(double x, double y, double z, ServerWorld serverWorld)
 	{
-		int m = (int)(getExplosionPower() / 2);
-		for (int i = 0; i < world.getPlayers().size(); ++i)
-		{
-			ServerPlayerEntity player = world.getPlayers().get(i);
-			world.spawnParticles(player, ParticleTypes.FLASH, true, x, y, z, 1, 0, 0, 0, 0);
-
-			world.spawnParticles(player, ParticleTypes.LARGE_SMOKE, true, x, y, z, 50 * m, 1, 1, 1, 0);
-			world.spawnParticles(player, ParticleTypes.SMOKE, true, x, y, z, 45 * m, 0.75 * m, 0.75 * m, 0.75 * m, 0);
+		float m = getExplosionPower() / 4;
```

Comment:
save the power to a variable and reference it on these 3 lines instead of calling the method 3 times